Gildner_Tyler/Assignments/ninjagold/ninjagold.py

```diff
@@ -0,0 +1,34 @@
+from flask import Flask, render_template, session, request, redirect
+import random
+
+app = Flask(__name__)
+app.secret_key = '123456'
+
+@app.route('/')
+def index():
+    if not 'gold' in session:
+        session['gold'] = 0
+    if not 'activities' in session:
+        session['activities'] = []
+    return render_template('index.html')
+
+@app.route('/process', methods = ['POST'])
+def process():
+    locations = {
```

Comment:
Efficiency: although this code works, you're running the random function for each location every time.  This means the cpu is doing 4 times the work that's needed.  Think about how you could rewrite this to only calculate the gold based on which location is clicked.  You'll get another chance at this in a future stack too (Ruby).